spec/support/shared_examples/address_transactions.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+shared_examples 'an address that supports transaction queries' do |_operation|
+  let(:external_addresses_api) { instance_double(Coinbase::Client::ExternalAddressesApi) }
+
+  before do
+    allow(Coinbase::Client::ExternalAddressesApi).to receive(:new).and_return(external_addresses_api)
+  end
+
+  describe '#transactions' do
+    let(:transaction) { build(:transaction_model, from_address_id: 'from_address') }
+    let(:response) do
+      Coinbase::Client::AddressTransactionList.new(
+        data: [
+          transaction
+        ]
+      )
+    end
+
+    before do
+      allow(external_addresses_api)
+        .to receive(:list_address_transactions)
+        .with(normalized_network_id, address_id, { limit: 10, page: nil })
+        .and_return(response)
+    end
+
+    context 'when list transactions' do
+      it 'returns the correct transactions' do
+        expect(address.transactions.first.from_address_id).to eq 'from_address'
+      end
+    end
+  end
+
+  describe Coinbase::Address do
+    describe '.transactions' do
```

Comment:
This is still a test for an instance method, this shared example should just be included in the `#transactions` describe block!